src/callout/promise.rs

```diff
@@ -0,0 +1,341 @@
+// Copyright 2024 Google LLC
```

Comment:
`std::future` in c++ is an async primitive to get result in multi-thread programming. `std::future` produced by `std::async` or `std::promise`. Once program calls `std::future::get`, it will block until `std::async` completed or `std::promise::set` called. It means at least two threads for provider and consumer but wasm has no (once call dispatched, thread will be used to process other requests).
And `future-rs` is providing the foundations for asynchronous programming in Rust, and lazy executed which requires an executor to poll the result (which means all operations must be non-blocking). The problem as same as`std::future`, there is no thread as executor to poll the result.
So the only way that we could own the thread is as callback of http response.